### PR TITLE
Add TypeScript type declarations

### PR DESCRIPTION
**Pull request by Mesoptier**

### CODE REVIEW

**Comment by deanilvincent** [comment]
Hi @Mesoptier !

Thank you for this contribution! If you don't mind, can you push to this branch [releases/v2.0.3](https://github.com/deanilvincent/check-password-strength/tree/releases/v2.0.3)? We'll include this on the upcoming updates/versions of our package. :) Stay safe!

**Comment by Mesoptier** [comment]
@deanilvincent Is this what you meant? If so, that should do it :)

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+export interface Option {
+  id: number;
+  value?: string;
```
Comment (Ennoriel) [inline index.d.ts:3]: value can be broader than a string (same line 21).

(I don't know how to asses in typescript that the value option parameter and the value attribute of the return object are of the same type)

**Review by Ennoriel — COMMENTED** [review]
Thanks a lot for your PR!

File: index.d.ts (hunk 1 above)
Comment (Mesoptier) [reply index.d.ts]: Good call! I've pushed an update that makes the type of `value` generic, so any kind of object can be used.

One issue is that the types now don't support omitting the `value` property. But I'm not sure if that even is officially supported behavior.

**Comment by deanilvincent** [comment]
Merging this PR! Thanks everyone & stay safe! 🥳

**Pull request merged**